src/Upgrade/Infrastructure/Processor/Strategy/Composer/Steps/IntegratorStep.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Upgrade\Infrastructure\Processor\Strategy\Composer\Steps;
+
+use Upgrade\Infrastructure\Dto\Step\StepsExecutionDto;
+use Upgrade\Infrastructure\Process\ProcessRunner;
+use Upgrade\Infrastructure\Processor\Strategy\RollbackStepInterface;
+use Upgrade\Infrastructure\VersionControlSystem\Adapter\Resolver\VersionControlSystemAdapterResolver;
+
+class IntegratorStep  extends AbstractStep implements RollbackStepInterface
+{
+
+    /**
+     * @var string
+     */
+    protected const RUNNER = '~/.composer/vendor/bin/integrator';
```

Comment:
To my mind better create a Separate service for access to the integrator. for example IntergatorClient. see ComposerLockComparator.php as example